content/apps/create-app-in-php.md

```diff
@@ -0,0 +1,129 @@
+immediately in this Activation URL.
+
+Let's create an `activate.php` file:
+```php
+
+const OAUTH_CLIENT_ID = '<CLIENT_ID>';
+const OAUTH_SCOPES = '<SCOPES>'; // eg: read_products
+
+session_start();
+
+$pimUrl = $_GET['pim_url'];
+if (empty($pimUrl)) {
+    exit('Missing PIM URL in the query');
+}
+
+$scheme = $_SERVER['HTTP_X_FORWARDED_PROTO'] ?: $_SERVER['REQUEST_SCHEME'];
+$host = $_SERVER['HTTP_HOST'];
+$redirectUri = sprintf('%s://%s/callback.php', $scheme, $host);
+// create a random state for preventing cross-site request forgery
+$state = bin2hex(random_bytes(10));
+
+// Store in the user session the state and the PIM URL
+$_SESSION['oauth2_state'] = $state;
+$_SESSION['pim_url'] = $pimUrl;
+
+// Build the parameters for the Authorization Request
+// https://datatracker.ietf.org/doc/html/rfc6749#section-4.1.1
+$authorizeUrlParams = http_build_query([
+    'response_type' => 'code',
+    'client_id' => OAUTH_CLIENT_ID,
+    'redirect_uri' => $redirectUri,
+    'scope' => OAUTH_SCOPES,
+    'state' => $state,
+]);
+
+// Build the url for the Authorization Request using the PIM URL
+$authorizeUrl = $pimUrl . '/connect/apps/v1/authorize?' . $authorizeUrlParams;
+
+// Redirect the user to the Authorization URL
+header('Location: ' . $authorizeUrl);
+```
+
+## Callback URL
+
+Then, your application must expose a callback URL.
+
+::: info
+The Code Challenge is documented [here](/apps/using-oauth2.html#whats-the-code-challenge).
+:::
+
+Let's create an `callback.php` file:
+```php
+
+const OAUTH_CLIENT_ID = '<CLIENT_ID>';
+const OAUTH_CLIENT_SECRET = '<CLIENT_SECRET>';
+
+session_start();
+
+// We check if the received state is the same as in the session, for security.
+$sessionState = $_SESSION['oauth2_state'] ?? '';
+$state = $_GET['state'] ?? '';
+if (empty($state) || $state !== $sessionState) {
+    exit('Invalid state');
+}
+
+$authorizationCode = $_GET['code'] ?? '';
+if (empty($authorizationCode)) {
+    exit('Missing authorization code');
+}
+
+$pimUrl = $_SESSION['pim_url'] ?? '';
+if (empty($pimUrl)) {
+    exit('No PIM url in session');
+}
+
+$codeIdentifier = bin2hex(random_bytes(30));
+$codeChallenge = hash('sha256', $codeIdentifier . OAUTH_CLIENT_SECRET);
+
+$accessTokenUrl = $pimUrl . '/connect/apps/v1/oauth2/token';
+$accessTokenRequestPayload = [
+    'client_id' => OAUTH_CLIENT_ID,
+    'code_identifier' => $codeIdentifier,
+    'code_challenge' => $codeChallenge,
+    'code' => $authorizationCode,
+    'grant_type' => 'authorization_code',
+];
+
+// Do a POST request on the token endpoint
+$ch = curl_init();
+curl_setopt($ch, CURLOPT_URL, $accessTokenUrl);
+curl_setopt($ch, CURLOPT_POST, true);
+curl_setopt($ch, CURLOPT_POSTFIELDS, $accessTokenRequestPayload);
+curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+$response = json_decode(curl_exec($ch), true);
+
+echo $response['access_token'];
+```
+
+And that's all. At the end of this process, you have now received the following response with an `access_token`:
```

Comment:
is there somewhere in the doc where you precise that the `access_token` has no TTL?